tests/consensus_tests/test_cluster_rejoin.py

```diff
@@ -73,3 +75,137 @@ def test_rejoin_cluster(tmp_path: pathlib.Path):
     wait_for_all_replicas_active(peer_api_uris[0], "test_collection2")
     # Assert that the restarted node has recovered the new collection
     wait_for_all_replicas_active(new_url, "test_collection2")
+
+
+def test_rejoin_origin_from_wal(tmp_path: pathlib.Path):
+    """
+    This test checks that origin peer (first peer of the cluster) commits its own peer ID to consensus.
+
+    - remove origin peer from cluster
+    - modify second peer's `raft_state.json`, so that it does *not* provide origin peer ID and URL
+      when bootstrapping new peer
+    - add new peer to the cluster (bootstrapping from second peer), and check that it has valid
+      state after it syncs with consensus
+    - if new peer has valid state at the end of the test, it means it received correct origin peer
+      ID and URL from consensus
+    """
+
+    # Overwrite `first_voter` peer
+    def overwrite_first_voter(state: dict[str, Any], _: Any):
+        state["first_voter"] = state["this_peer_id"]
+        return state
+
+    rejoin_cluster_test(tmp_path, start_cluster, overwrite_first_voter)
+
+
+def rejoin_cluster_test(
+    tmp_path: pathlib.Path,
+    start_cluster: Callable[[pathlib.Path, int], tuple[list[str], list[pathlib.Path], str]],
+    raft_state: Callable[[dict[str, Any], int], Any | None],
+    collection: str = "test_collection",
+    peers: int = 3,
+    shards: int = 3,
+    expected_shards: int = 3,
+):
+    """
+    Parameterized test body, that tests adding new peer after origin peer was removed from the cluster.
```

Comment:
Not necessary, but maybe we can link the issue here because this roughly implements the described repro.

```suggestion
    Parameterized test body, that tests adding new peer after origin peer was removed from the cluster.
    See: <https://github.com/qdrant/qdrant/issues/5138>
```

Reply:
Sure!